usr/src/cmd/utmpd/utmpd.c

```diff
@@ -301,6 +354,8 @@ main(int argc, char *argv[])
 	if ((WTMPXfd = open(WTMPX_FILE, O_RDONLY)) < 0)
 		nonfatal("WARNING: unable to open " WTMPX_FILE " for update.");
 
+	door_server_init();
```

Comment:
Since in this implementation the global zone instance of `utmpd` is responsible for fixing up all the zone, it's the only one that neds to set up the door.

```suggestion
	if (getzoneid() == GLOBAL_ZONEID)
		door_server_init();
```

Reply:
> Perhaps the kernel should call up to each zone's `utmpd` and get them to fix up themselves? That would remove the dependency on libzonecfg and simplify `utmpd` a bit, at the expense of a bit more work in the kernel to iterate the zones and build up the door path for each.

Yeah, this really sounds like the preferred solution. I'll put the PR on draft for now until this is implemented.